servicetalk-http-utils/src/test/java/io/servicetalk/http/utils/AbstractTimeoutHttpFilterTest.java

```diff
@@ -108,23 +128,30 @@ private void responseWithNonPositiveTimeout(Duration timeout, boolean fullReques
     @ValueSource(booleans = {false, true})
     void responseCompletesBeforeTimeout(boolean fullRequestResponse) {
         TestSingle<StreamingHttpResponse> responseSingle = new TestSingle<>();
-        StepVerifiers.create(applyFilter(ofSeconds(DEFAULT_TIMEOUT_SECONDS / 2), fullRequestResponse, responseSingle))
+        StepVerifiers.create(applyFilter(ofSeconds(DEFAULT_TIMEOUT_SECONDS / 2),
+                        fullRequestResponse, defaultStrategy(), responseSingle))
                 .then(() -> immediate().schedule(() -> responseSingle.onSuccess(mock(StreamingHttpResponse.class)),
                         ofMillis(50L)))
                 .expectSuccess()
                 .verify();
         assertThat("No subscribe for response single", responseSingle.isSubscribed(), is(true));
     }
 
-    @Test
-    void payloadBodyTimeout() {
+    static Iterable<HttpExecutionStrategy> executionStrategies() {
+        return Arrays.asList(offloadNever(), offloadNone(), defaultStrategy(), offloadAll());
```

Comment:
Consider adding a custom-built execution strategy to the test

Reply:
Yes, I agree this is important addition.